### PR TITLE
Use assisted installer admin kubeconfig if exists

### PR DESCRIPTION
**Pull request by jhernand**
Currently the deployment manager service uses the admin kubeconfig that was used to register the cluster with ACM. But this will not exist if the cluster was created with assisted installer. To avoid that problem this patch changes the service so that it first check if the secret used by the assisted installer exists. If it does exist then it will use it.

### CODE REVIEW

File: internal/service/deployment_manager_handler.go [hunk 1]
```diff
@@ -579,8 +579,20 @@ func (h *DeploymentManagerHandler) fetchProfile(ctx context.Context,
...
+	assistedInstallerAdminKubeconfig, err := h.fetchAssistedInstallerAdminKubeconfig(
+		ctx,
+		h.hubClient,
+		cluster,
+	)
+	if err != nil {
+		return
+	}
+	if assistedInstallerAdminKubeconfig != nil {
+		result, err = h.makeProfile(assistedInstallerAdminKubeconfig)
```
Comment (danielerez) [inline internal/service/deployment_manager_handler.go:593]: Maybe worth having some log here?
I.e. a debug/info log that indicates which kubeconfig is used.

**Review by danielerez — COMMENTED** [review]
Looks good, just a minor comment.

File: internal/service/deployment_manager_handler.go (hunk 1 above)
Comment (jhernand) [reply internal/service/deployment_manager_handler.go:593]: Makes sense, will do it.

File: internal/service/deployment_manager_handler.go [hunk 2]
```diff
@@ -638,12 +654,58 @@ func (h *DeploymentManagerHandler) fetchProfile(ctx context.Context,
...
+	)
 
 	// Make the profile data from the cluster admin kubeconfig:
 	result, err = h.makeProfile(clusterAdminKubeconfig)
 	return
 }
 
+// fetchAssistedInstallerAdminKubeconfig uses the given Kubernetes API client to fetch the admin
+// kubeconfig that is created by the assisted installer when cluster installation finishes. It
+// returns the serialized kubeconfig, or nil if there it doesn't exist, for example if the cluster
```
Comment (irinamihai) [inline internal/service/deployment_manager_handler.go:669]: Typo: extra "there".

Comment (jhernand) [reply internal/service/deployment_manager_handler.go:669]: Thanks Irina :-) , fixing it ...

**Pull request merged**